components/script/dom/range.rs

```diff
@@ -1337,109 +1337,93 @@ impl WeakRangeVec {
-                    range.end().set_offset(offset + 1);
-                }
-            });
-        }
+        self.cell.borrow_mut().update(|entry| {
+            let range = entry.root().unwrap();
+            if range.start().node() == node && offset == range.start_offset() {
+                range.report_change();
+                range.start().set_offset(offset + 1);
+            }
+            if range.end().node() == node && offset == range.end_offset() {
+                range.report_change();
+                range.end().set_offset(offset + 1);
+            }
+        });
     }
 
     fn map_offset_above<F: FnMut(u32) -> u32>(&self, node: &Node, offset: u32, mut f: F) {
-        unsafe {
-            (*self.cell.get()).update(|entry| {
-                let range = entry.root().unwrap();
-                let start_offset = range.start_offset();
-                if range.start().node() == node && start_offset > offset {
-                    range.report_change();
-                    range.start().set_offset(f(start_offset));
-                }
-                let end_offset = range.end_offset();
-                if range.end().node() == node && end_offset > offset {
-                    range.report_change();
-                    range.end().set_offset(f(end_offset));
-                }
-            });
-        }
+        self.cell.borrow_mut().update(|entry| {
+            let range = entry.root().unwrap();
+            let start_offset = range.start_offset();
+            if range.start().node() == node && start_offset > offset {
+                range.report_change();
+                range.start().set_offset(f(start_offset));
+            }
+            let end_offset = range.end_offset();
+            if range.end().node() == node && end_offset > offset {
+                range.report_change();
+                range.end().set_offset(f(end_offset));
+            }
+        });
     }
 
     pub(crate) fn push(&self, ref_: WeakRef<Range>) {
-        unsafe {
-            (*self.cell.get()).push(ref_);
-        }
+        self.cell.borrow_mut().push(ref_);
     }
 
     fn remove(&self, range: &Range) -> WeakRef<Range> {
-        unsafe {
-            let ranges = &mut *self.cell.get();
-            let position = ranges.iter().position(|ref_| ref_ == range).unwrap();
-            ranges.swap_remove(position)
-        }
-    }
-}
-
-#[allow(unsafe_code)]
-impl MallocSizeOf for WeakRangeVec {
-    fn size_of(&self, ops: &mut MallocSizeOfOps) -> usize {
-        unsafe { (*self.cell.get()).size_of(ops) }
+        let mut ranges = self.cell.borrow_mut();
+        let position = ranges.iter().position(|ref_| ref_ == range).unwrap();
+        ranges.swap_remove(position)
     }
 }
 
 #[allow(unsafe_code)]
 unsafe impl JSTraceable for WeakRangeVec {
     unsafe fn trace(&self, _: *mut JSTracer) {
-        (*self.cell.get()).retain_alive()
+        self.cell.borrow_mut().retain_alive()
     }
 }
```

Comment:
Per documentation of [WeakRefVec](https://doc.servo.org/script/dom/bindings/weakref/struct.WeakRefVec.html), this should already be default tracing impl of WeakRefVec, although I did not find it's trace function. Then there is also [`DOMTracker`](https://doc.servo.org/script/dom/bindings/weakref/struct.DOMTracker.html) which is wrapper for `DomRefCell<WeakRefVec<T>>`.